velox/expression/SwitchExpr.cpp

```diff
@@ -39,9 +40,38 @@ SwitchExpr::SwitchExpr(
       hasElseClause_{hasElseClause(inputs_)} {
   VELOX_CHECK_GT(numCases_, 0);
 
+  const auto& thenType = inputs_[1]->type();
+
   for (auto i = 0; i < numCases_; i++) {
     auto& condition = inputs_[i * 2];
     VELOX_CHECK_EQ(condition->type()->kind(), TypeKind::BOOLEAN);
+    auto& thenClause = inputs_[i * 2 + 1];
+    if (thenClause->type()->isUnKnown()) {
+      // Allow null constants.
+      VELOX_CHECK_NOT_NULL(dynamic_cast<ConstantExpr*>(thenClause.get()));
```

Comment:
Curious: when do we have null constants for then clause ?

Reply:
For example, if (a > 5, null, 10).